### PR TITLE
Added x and y values to the geometries property of cb_data for TapTool

### LINKED ISSUE #8013: Expose the more detailed geometry data to the tap tool
When a callback is written for the tap_tool, it exposes the screen geometry of the point that was tapped, available as 
```
cb_data["geometries"]["sx"]
cb_data["geometries"]["sy"]
```
However, to be truly useful it should also provide the converted data coordinates as 
```
cb_data["geometries"]["x"]
cb_data["geometries"]["y"]
```
 This would emulate the hover tool functionality. (Apart from calling one geometry and the other geometries)

There are many cases where it is useful to know exactly what the data values are when a point is tapped. On a line, the point tapped may be somewhere between two distant points, and we need the extrapolated values for that exact point along the line.

Currently, the only way to get these values is to hack an additional callback on the hover tool, and save the values in a CDS so we can interrogate them during the tap tool callback. This is very expensive as we are using the hover event to continually store this data even if we never hit the tap tool.

Below is a current example of this hack:
```
# Create a CDS so we can store the hover x and y coordinates.
locator = ColumnDataSource(name="pred_locator", data=dict(x=[0], y=[0]))

# Define a call back for the hover tool
callback_hover = CustomJS(args=dict(locator=locator), code="""
         geometry = cb_data["geometry"];
         locator.data["x"][0] = geometry["x"];
         locator.data["y"][0] = geometry["y"];
         """)

# define a callback for the tap tool
callback_tap = CustomJS(args=dict(locator=locator), code="""
         geometry = cb_data["geometries"];

         var y_data_val = locator.data["y"][0];
         var y_screen_val = geometry["sy"]; 
         alert("Tapped at " + y_screen_val + ", actual value is " + y_data_val);
         """)


plot.add_tools(HoverTool(callback=callback_hover))
plot.add_tools(TapTool(callback=callback_tap))

```
This is related to the original request to expose the geometry to the tap tool.

Expose Geometries to TapTool Callbacks #4082

### PR DESCRIPTION
**Pull request by euandekock**
The Tap Tool doesn't expose any direct access to the calculated range-specific x and y data coordinates. This is available in all the Select tools, and in the Hover tool. 

The method used here is similar to the approach used in the Select and Hover tools in that we step outside the Geometry classes and define a naked object to hold all the values. Most user-side js code will not reference or understand these internal classes and will access the values directly.

All pull requests must have an associated issue in the issue tracker. If there
isn't one, please go open an issue describing the defect, deficiency or desired
feature. You can read more about our issue and PR processes in the
[wiki](https://github.com/bokeh/bokeh/wiki/BEP-1:-Issues-and-PRs-management).

- [x] issues: fixes #8013 
- [ ] tests added / passed
- [ ] release document entry (if new feature or API change)

### CODE REVIEW

**Comment by euandekock** [comment]
@mattpap, please can you review this as per the request from @bryevdv . I'd like to get it in the code to at least provide feature parity between TapTool and the Select and Hover tools, but I'm also open to looking at how we can extend the event subsystem to replace the callback functions.

**Comment by euandekock** [comment]
@mattpap and @brydev FYI I cannot access the gitter discussion forum at work due to network restrictions, so can we keep the discussions in the issue or in the PR. Thanks.

**Comment by euandekock** [comment]
@mattpap , @bryevdv Just following up to see if there is any progress on this? I'd really like to get it completed as I've finished my project that justified my work on it for now.

I'm happy to revisit the broader events subsystem when my new work reaches a stage where I I am back on visualisation work.

File: bokehjs/src/lib/models/tools/gestures/tap_tool.ts [hunk 1]
```diff
@@ -40,7 +32,16 @@ export class TapToolView extends SelectToolView {
...
         if (did_hit && callback != null) {
-          cb_data.source = sm.source
+          const frame = this.plot_model.frame
+          const xscale = frame.xscales[renderers[0].x_range_name]
+          const yscale = frame.yscales[renderers[0].y_range_name]
+          const x = xscale.invert(geometry.sx)
+          const y = yscale.invert(geometry.sy)
+          const g = {...geometry, x, y}
+          const cb_data: {
+              geometries: { type: "point", sx: number, sy: number, x: number, y: number }
```
Comment (mattpap) [inline bokehjs/src/lib/models/tools/gestures/tap_tool.ts]: This would be more meaningful if the type was `PointGeometry & {x: number, y: number}`.

**Comment by euandekock** [comment]
OK, thanks for the suggestion - my understanding of TypeScript is limited, so I didn't realise you could do that. Changes made as requested.

**Pull request merged**